server/src/main/java/io/crate/execution/engine/aggregation/impl/average/numeric/NumericAverageState.java

```diff
@@ -44,7 +45,10 @@ public class NumericAverageState<T extends NumericValueHolder> implements Compar
 
     public BigDecimal value() {
         if (count > 0) {
-            return sum.value().divide(BigDecimal.valueOf(count));
+            // We need to use divide with a MathContext to avoid ArithmeticException on infinite fractions.
+            // Using best (finite) precision as we want to compute final result with the best precision.
+            // If users want to cut final result, they can use explicit cast.
+            return sum.value().divide(BigDecimal.valueOf(count), MathContext.DECIMAL128);
```

Comment:
Why not use `UNLIMITED` here? si there a difference?

Reply:
UNLIMITED would hit ArithemticException. 

Also, that's the reason why using bound `returnType` would not only lose precision but also keep the issue.
Default NumericType instance for non-stored values (for example literals) has null scale and precision, so MatchContext obtained from the default NumericType would be `UNLIMITED` and hit ArithmeticException

Reply:
thx!